src/kademlia/traversal_algorithm.cpp

```diff
@@ -593,6 +593,24 @@ void traversal_algorithm::status(dht_lookup& l)
 	l.last_sent = last_sent;
 }
 
+void traversal_algorithm::look_for_nodes(bdecode_node const& r, std::function<void(const node_endpoint&)> f)
+{
+	udp const protocol = get_node().protocol();
+	int const protocol_size = int(detail::address_size(protocol));
+	char const* nodes_key = get_node().protocol_nodes_key();
```

Comment:
I really think this should be a parameter instead of the whole function being a member function. (i.e. ``nodes_key``)